debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/DebeziumEngineIT.java

```diff
@@ -197,4 +198,61 @@ public void shouldSerializeToCloudEvents() throws Exception {
         }
     }
 
+    @Test
+    @FixFor("DBZ-2461")
+    public void testOffsetsCommitAfterStop() throws Exception {
+        final AtomicReference<Throwable> exception = new AtomicReference<>();
+        DebeziumEngine<ChangeEvent<String, String>> engine;
+
+        TestHelper.execute("DROP TABLE IF EXISTS tests;", "CREATE TABLE tests (id SERIAL PRIMARY KEY);");
+
+        final Properties props = new Properties();
+        props.putAll(TestHelper.defaultConfig().build().asMap());
+        props.setProperty("name", "debezium-engine");
+        props.setProperty("connector.class", "io.debezium.connector.postgresql.PostgresConnector");
+        props.setProperty(StandaloneConfig.OFFSET_STORAGE_FILE_FILENAME_CONFIG,
+                OFFSET_STORE_PATH.toAbsolutePath().toString());
+        props.setProperty("offset.flush.interval.ms", "3000");
+        props.setProperty("converter.schemas.enable", "false");
+
+        engine = DebeziumEngine.create(Json.class).using(props).using(new DebeziumEngine.ConnectorCallback() {
+            @Override
+            public void connectorStarted() {
+            }
+
+            @Override
+            public void connectorStopped() {
+            }
+        }).using((success, message, error) -> {
+            exception.compareAndSet(null, error);
+        }).notifying((records, committer) -> {
+            try {
+
+                for (ChangeEvent<String, String> record : records) {
+                    committer.markProcessed(record);
+                }
+                committer.markBatchFinished();
+            }
+            catch (Exception e) {
+                Testing.printError(e);
+            }
+        }).build();
+
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        executor.execute(engine);
+
+        for (int i = 0; i < 100; i++) {
+            TestHelper.execute("INSERT INTO tests VALUES(default)");
+        }
+        Thread.sleep(5000);
```

Comment:
Can we avoid this via Awaitility?

Reply:
Probably yes but I am not sure if it is worth the effort. This is just a way how to increase the probaility that the bug is visible, not for the test stability. TO use awaitility we would need to monitor offset store and validate if there are offsets of some condition.

Reply:
I'm not so much concerned about stability, but about overall execution time of the test suite.

Reply:
We are not going to short it down much anyway. We need to wait for 3 secs at least to make sure the offset committer commits offsets. I don't want to reduce that interval too much as with loer values the issue tended to disappear.

Reply:
Ok, not super-excited, but so be it.

Reply:
Removed as by-product of fixing block issue